core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -110,10 +112,9 @@
     return this;
   }
 
-  @Override
-  public void start(Listener<RespT> observer, Metadata headers) {
-    Preconditions.checkState(stream == null, "Already started");
-
+  @VisibleForTesting
+  static void prepareHeaders(Metadata headers, CallOptions callOptions, String userAgent,
```

Comment:
Why move it to a util class? Shouldn't it only be used this one place, and some of the other behavior of this class is co-dependent with the behavior of this method.